src/main/java/org/opentripplanner/util/OTPFeature.java

```diff
@@ -0,0 +1,118 @@
+package org.opentripplanner.util;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * The purpose of this class is to be able to turn features on and off.
+ * <p/>
+ * This configuration is optional an found under "feature" in the top
+ * level 'otp-config.json' file.
+ */
+public enum OTPFeature {
+    APIExternalGeocoder(true),
+    APIBikeRental(true),
+    APIAlertPatcher(true),
+    APIRouters(true),
+    APIServerInfo(true),
+    APIGraphInspectorTile(true),
+    APIUpdaterStatus(true);
+
+    private static final String VALUE_ON = "on";
+    private static final String VALUE_OFF = "off";
+    private static final String VALUE_ENABLE = "enable";
+    private static final String VALUE_DISABLE = "disable";
+
+    private static final Logger LOG = LoggerFactory.getLogger(OTPFeature.class);
+
+    OTPFeature(boolean defaultEnabled) {
+        this.enabled = defaultEnabled;
+    }
+
+    private boolean enabled;
+
+    /**
+     * Return {@code true} if feature is turned 'on'.
+     */
+    public boolean on() {
+        return enabled;
+    }
+
+    /**
+     * Return {@code true} if feature is turned 'off'.
+     */
+    public boolean off() {
+        return !enabled;
+    }
+
+    /**
+     * Allow unit test and this class to enable/disable a feature.
+     */
+    void set(boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    /**
+     * Return "on" if feature is enabled, if not return "off".
+     */
+    public String valueAsString() {
+        return enabled ? VALUE_ON : VALUE_OFF;
+    }
+
+    /**
+     * Configure features using given JSON.
+     */
+    public static void configure(@NotNull JsonNode otpConfig) {
+        JsonNode features = otpConfig.path("features");
+        for (OTPFeature feature : values()) {
+            setFeatureFromConfig(feature, features.path(feature.name()));
+        }
+    }
+
+
+    /* private members */
+
+    private static void setFeatureFromConfig(OTPFeature feature, JsonNode node) {
+        if (!node.isMissingNode()) {
+            boolean featureValue = isOn(node.asText(feature.valueAsString()), feature.name());
+            if (featureValue != feature.enabled) {
+                feature.set(featureValue);
+                logFeature(feature);
+            } else {
+                logFeatureDefault(feature);
+            }
+        } else {
+            logFeatureDefault(feature);
+        }
+    }
+
+    private static void logFeature(OTPFeature feature) {
+        LOG.info("Feature '{}' is '{}'.", feature.name(), feature.valueAsString());
+    }
+
+    private static void logFeatureDefault(OTPFeature feature) {
+        LOG.info("Feature '{}' is '{}' (default).", feature.name(), feature.valueAsString());
+    }
+
+    private static boolean isOn(String value, String key) {
+        if (match(value, VALUE_ON) || match(value, VALUE_ENABLE)) {
```

Comment:
We decided not to accept multiple equivalent values (yes/enable). In fact we could just make these all boolean values inside a featuresEnabled object.

Reply:
I allowed only `on` and `off`. I think it is easier to keep a list of features with a value `on`/`off`, than two list, one with enabled features and one with disabled features. This would require the feture to be copied back an forth between these to lists (or deleted). We need both list because some features are enabled by default and som is not.

Reply:
I don't think we would need two lists. Just a single list with different default values for each feature. Whether or not they are text strings "on"/"off", "enabled"/"disabled", or boolean true/false should have no effect. For example:

```
featuresEnabled: {
  // disable the ServerInfo API, which is enabled by default
  APIServerInfo: false, 
  // enable the Graph Statistics API which is disabled by default
  SandboxExampleAPIGraphStatistics: true 
}
```

Reply:
Separate point: the conventional order of parameters would be (key, value) rather than (value, key) right? Does this reversal have a significance here?

Reply:
So this is how it currently looks like, I guess that is almost what you have above. I can change the label to "featuresEnabled":
```
{
    "features" : {
        APIAlertPatcher : "off",
        APIBikeRental : "off",
        APIExternalGeocoder : "off",
        APIGraphInspectorTile : "on",
        APIRouters : "on",
        APIServerInfo : "off",
        APIUpdaterStatus : "off"
    }
}

```